script/tool/lib/src/analyze_command.dart

```diff
@@ -64,6 +65,14 @@ class AnalyzeCommand extends PluginCommand {
     }
 
     final List<Directory> packageDirectories = await getPackages().toList();
+    final List<String> packagePaths =
+        packageDirectories.map((Directory dir) => dir.path).toList();
+    packageDirectories.removeWhere((Directory directory) {
+      // We remove the 'example' subdirectories - 'flutter pub get' automatically
+      // runs 'pub get' there as part of handling the parent directory.
+      return directory.basename == 'example' &&
+          packagePaths.contains(directory.parent.path);
+    });
```

Comment:
I was hoping we could just use `getPlugins` instead of doing the filtering, but it looks like flutter/packages has some other nested constructions.

Reply:
Yeah, I would prefer something more semantic here as well. The logic for how these are computed - getPlugins() and getPackages() - seems complicated, and I wasn't comfortable modifying it.

It looks like `plugins` actually returns packages? And `packages` returns packages plus there example directories?

I would have assumed that `getPlugins()` would return the directory containing the n dart packages that make up a plugin - like `url_launcher/`, `url_launcher_platform_interface/`, `url_launcher_macos/`, ...

I'll want a call like that - the top level plugin directory - for a follow up PR, where I'm going to try and speed up the analysis portion of the analyze call.

Reply:
> It looks like plugins actually returns packages? And packages returns packages plus there example directories?

I suspect what happened is that this was written for flutter/plugins, then expanded for use in flutter/packages without renaming. Fixing the fact that "plugins" is used to refer to packages in general in the tool (including the `--plugins` argument) is on my list.